patches/server/0016-Add-configurable-despawn-distances-for-living-entiti.patch

```diff
@@ -5,51 +5,75 @@ Subject: [PATCH] Add configurable despawn distances for living entities
 
 
 diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
-index 419f3bca4c1dd18790ad9e602bdec009c2933606..4634da27cde654e682ac4525df9850ea40afbb87 100644
+index 8077c9016d0119ddba8755c839be231ab19b6ccc..8399d7d845482ce3eaedb1e6419f59e71925e676 100644
 --- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
 +++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
-@@ -107,4 +107,20 @@ public class PaperWorldConfig {
+@@ -1,7 +1,14 @@
+ package com.destroystokyo.paper;
+ 
++import java.util.Arrays;
++import java.util.EnumMap;
+ import java.util.List;
++import java.util.Map;
++import java.util.function.Function;
++import java.util.stream.Collector;
++import java.util.stream.Collectors;
+ 
++import net.minecraft.world.entity.MobCategory;
+ import org.bukkit.Bukkit;
+ import org.bukkit.configuration.file.YamlConfiguration;
+ import org.spigotmc.SpigotWorldConfig;
+@@ -113,4 +120,29 @@ public class PaperWorldConfig {
      private void nerfedMobsShouldJump() {
          nerfedMobsShouldJump = getBoolean("spawner-nerfed-mobs-should-jump", false);
      }
 +
-+    public int softDespawnDistance;
-+    public int hardDespawnDistance;
++    public final Reference2IntMap<MobCategory> softDespawnDistances = Arrays.stream(MobCategory.values()).collect(Collector.of(() -> new Reference2IntOpenHashMap<MobCategory>(MobCategory.values().length), (map, category) -> map.put(category, category.getNoDespawnDistance()), (map1, map2) -> { map1.putAll(map2); return map1; }));
```

Comment:
Now the map init looks even more cursed (+_+;)

Reply:
well hey, you asked for the FU map. It should be a map too, not fields for each one.

Reply:
ya ok. I think I got what you meant now.